db/migrate/20180828161400_remove_join_table_for_role_and_usage.rb

```diff
@@ -0,0 +1,14 @@
+class RemoveJoinTableForRoleAndUsage < ActiveRecord::Migration[5.1]
+  def change
+    remove_foreign_key :katello_subscription_facet_purpose_roles, :katello_subscription_facets
+    remove_foreign_key :katello_subscription_facet_purpose_roles, :katello_purpose_roles
+    remove_foreign_key :katello_subscription_facet_purpose_usages, :katello_subscription_facets
+    remove_foreign_key :katello_subscription_facet_purpose_usages, :katello_purpose_usages
+
```

Comment:
How about a unique constraint on SubscriptionFacetPurposeAddon purpose_addon_id and subscription_facet_id?